### ISSUE TITLE
Use propper logging

### ISSUE THREAD

**Issue by schnecki** [mr/ftp-client#14]
It would make sense to use a proper logging mechanism instead of writing to standard output. I am thinking of `Control.Monad.Logger` or similar.
Otherwise at least add a preprocessor statement for turning on/off debugging:

    debugging = 
    #ifdef DEBUG
        True       
    #else 
        False
    #endif

**Comment by mr** [comment]
I'd definitely prefer to add real logging over CPP, but it's a matter of time for me right now. There are tons of things I'd like to improve, but other projects have just been more important to me lately.

I'm thinking about wrapping everything into a reader monad on top of IO that uses the handler as it's environment (or at least the handler will be part of it) and then instancing classes in monad-logger. It'll take some time to research and implement, and you're welcome to give it a shot if you want to send a PR.

**Comment by schnecki** [comment]
Yeah I am quite busy currently myself. Just wanted to list it as a issue, as I did not find it yet. If at some point I have time for it I might give it a try, but unfortunately that can take several months.
